src/LanguageServer/Protocol/Features/Options/SolutionAnalyzerConfigOptionsUpdater.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Options;
+
+/// <summary>
+/// Keeps <see cref="SolutionState.FallbackAnalyzerOptions"/> up-to-date with global option values maintained by <see cref="IGlobalOptionService"/>.
+/// </summary>
+[Export]
+[ExportEventListener(WellKnownEventListeners.Workspace,
+    [WorkspaceKind.Host, WorkspaceKind.Interactive, WorkspaceKind.SemanticSearch, WorkspaceKind.MetadataAsSource, WorkspaceKind.MiscellaneousFiles, WorkspaceKind.Debugger, WorkspaceKind.Preview]), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class SolutionAnalyzerConfigOptionsUpdater(IGlobalOptionService globalOptions) : IEventListener<object>, IEventListenerStoppable
+{
+    public void StartListening(Workspace workspace, object serviceOpt)
+        => globalOptions.AddOptionChangedHandler(workspace, GlobalOptionsChanged);
+
+    public void StopListening(Workspace workspace)
+        => globalOptions.RemoveOptionChangedHandler(workspace, GlobalOptionsChanged);
+
+    private void GlobalOptionsChanged(object sender, object target, OptionChangedEventArgs args)
+    {
+        Debug.Assert(target is Workspace);
+
+        try
+        {
+            if (!args.ChangedOptions.Any(static o => o.key.Option.Definition.IsEditorConfigOption))
```

Comment:
doc this.  it's not immediately apparent to me why we only care editorconfig options here.